uefi/src/data_types/owned_strs.rs

```diff
@@ -47,9 +50,66 @@ impl core::error::Error for FromStrError {}
 /// let s = CString16::try_from("abc").unwrap();
 /// assert_eq!(s.to_string(), "abc");
 /// ```
-#[derive(Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
 pub struct CString16(Vec<Char16>);
 
+impl CString16 {
+    /// Creates a new empty string with a null-byte.
+    #[must_use]
+    pub fn new() -> Self {
+        Self(vec![NUL_16])
+    }
+
+    /// Pushes a character to the end of the string.
+    ///
+    /// # Panics
+    /// Panics if the char is a null byte.
+    pub fn push(&mut self, char: Char16) {
+        assert_ne!(char, NUL_16, "Pushing a null-byte is illegal");
+        let last_elem = self
+            .0
+            .last_mut()
+            .expect("There should be at least a null byte");
+        *last_elem = char;
+        self.0.push(NUL_16);
+    }
+
+    /// Pushes a string to the end of the string.
+    ///
+    /// # Panics
+    /// Panics if the char is a null byte.
+    pub fn push_str(&mut self, str: &CStr16) {
+        str.as_slice()
+            .iter()
+            .copied()
+            .for_each(|char| self.push(char));
+    }
+
+    /// Replaces all chars in the string with the replace value in-place.
+    pub fn char_replace_all_in_place(&mut self, search: Char16, replace: Char16) {
+        let slice: &mut CStr16 = self.as_mut();
+        slice.char_replace_all_in_place(search, replace);
+    }
+
+    /// Returns the number of characters without the trailing null.
+    #[must_use]
+    pub fn len(&self) -> usize {
```

Comment:
How about `num_chars`? I don't think `len` is bad exactly, but I feel like I would always be looking it up to remind myself if it's bytes or chars and whether it includes the null char. With `num_chars` I'd only have to remember whether it includes the null or not.